nncf/torch/quantization/algo.py

```diff
@@ -502,18 +500,32 @@ def __init__(self, config, should_init: bool = True):
             self._use_logarithm_scale_per_group[quantizer_group] = params_dict.get('logarithm_scale', False)
 
         self._disable_saturation_fix = self.config.get('disable_saturation_fix', False)
+        self._single_config_quantizer_setup = None  # type: Optional[SingleConfigQuantizerSetup]
 
-    def _force_manual_precision_init(self):
-        init_config = self.config.get('initializer', {})
-        init_precision_config = init_config.get('precision', None)
-        if init_precision_config is not None:
-            precision_init_type = init_precision_config.get('type', 'manual')
-            if precision_init_type == 'manual':
-                # range init is needed for correct setting of Adjust Padding ops as it considers sign of FQ
-                self._range_init_params = self._parse_range_init_params(init_config)
-                self.should_init = True
-                self._precision_init_type = precision_init_type
-                self._precision_init_params = ManualPrecisionInitParams.from_config(init_precision_config)
+    def get_state(self) -> Dict[str, object]:
+        """
+        Returns a JSON-compatible dictionary containing a state of the object.
+        """
+        build_time_metric_infos_state = {}
+        if self._build_time_metric_infos:
+            build_time_metric_infos_state = self._build_time_metric_infos.get_state()
+        quantizer_setup_state = {}
+        if self._single_config_quantizer_setup:
+            quantizer_setup_state = self._single_config_quantizer_setup.get_state()
+        return {
+            self._QUANTIZER_SETUP_STATE_ATTR: quantizer_setup_state,
+            self._BUILD_TIME_METRIC_INFOS_STATE_ATTR: build_time_metric_infos_state
+        }
+
+    def load_state(self, state: Dict):
```

Comment:
This is probably related to my other comments, but still - should we introduce class-specific, well-typed structs for each stateful class (which will encapsulate the expected dict key constants) with a common `ISerializable` interface?

Reply:
it would contradict to the current approach with get_state, load_state design, when we expect dictionary with python objects.
moreover, for PT we don't need this states to be serialized to JSON. torch.save works fine with Python objects.
@alexsu52 @andrey-churkin